iceoryx_posh/test/moduletests/test_roudi_cmd_line_parser_config_file_option.cpp

```diff
@@ -97,10 +103,65 @@ TEST_F(CmdLineParserConfigFileOption_test, HelpLongOptionLeadsProgrammNotRunning
     args[0] = &appName[0];
     args[1] = &option[0];
 
-    iox::config::CmdLineParserConfigFileOption sut;
+    CmdLineParserConfigFileOption sut;
     auto result = sut.parse(numberOfArgs, args);
 
-    EXPECT_THAT(result.has_error(), Eq(false));
-    EXPECT_THAT(result.value().run, Eq(false));
+    EXPECT_FALSE(result.has_error());
+    EXPECT_FALSE(result.value().run);
 }
+
+TEST_F(CmdLineParserConfigFileOption_test, WrongOptionLeadsUnkownOptionResult)
+{
+    constexpr uint8_t numberOfArgs{2U};
+    char* args[numberOfArgs];
+    char appName[] = "./foo";
+    char option[] = "--unknown";
+    args[0] = &appName[0];
+    args[1] = &option[0];
+
+    CmdLineParserConfigFileOption sut;
+    auto result = sut.parse(numberOfArgs, args);
+
+    EXPECT_TRUE(result.has_error());
+    EXPECT_THAT(result.get_error(), Eq(CmdLineParserResult::UNKNOWN_OPTION_USED));
+}
+
+TEST_F(CmdLineParserConfigFileOption_test, UnknownOptionLeadsCallingCmdLineParserParseReturningNoError)
+{
+    constexpr uint8_t numberOfArgs{3U};
+    char* args[numberOfArgs];
+    char appName[] = "./foo";
+    char option[] = "-u";
+    char value[] = "4242";
+    args[0] = &appName[0];
+    args[1] = &option[0];
+    args[2] = &value[0];
+
+    CmdLineParserConfigFileOption sut;
+    auto result = sut.parse(numberOfArgs, args);
+
+    EXPECT_FALSE(result.has_error());
+    EXPECT_TRUE(result.value().uniqueRouDiId.has_value());
+    EXPECT_THAT(result.value().uniqueRouDiId.value(), Eq(4242));
+}
+
+TEST_F(CmdLineParserConfigFileOption_test, CmdLineParsingModeEqualToOneReturnNoError)
```

Comment:
When I read this title I assume that as long as you provide only one parameter it will succeed. Like `./foo --myFunkyOptionBlaBlub` will succeed even though it was not registered?!

Reply:
Not really, only one option should be taken in consideration if several are provided. Now, it makes me think that I should pass 2 options and make sure only the first one is used.

Reply:
I think what @elfenpiff means is to make the test name more precise, e.g. ``CmdLineParsingModeOneAndValidOptionReturnsNoError``.

Reply:
Looks like it's done.